internal/host/validator.go

```diff
@@ -393,11 +393,9 @@ func (v *validator) diskEncryptionRequirementsSatisfied(c *validationContext) Va
 		return ValidationPending
 	}
 
-	role := common.GetEffectiveRole(c.host)
```

Comment:
Was EnableOn was explicitly set to nil? Otherwise it would skip the flow in line 388.
It should set to default value ('none') here: https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L407
Maybe we should had 'nil' check to the validation as well then.

Reply:
This is day1 scenario. On day2 the EnableOn is nil because the entire structure is not initialized. With this, you want the validation to run still